### PR TITLE
Add tests for POST endpoint

### PR DESCRIPTION
**Pull request by gregzealley**

### CODE REVIEW

File: src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java [hunk 1]
```diff
@@ -20,10 +23,48 @@ public void setup() {
 
     @Test
     public void WhenCallPostsEndpointThenSuccessStatusReturned() {
```
Comment (oliverCausley) [inline src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java]:
```suggestion
    public void whenCallPostsEndpointThenSuccessStatusReturned() {
```

File: src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java [hunk 2]
```diff
@@ -20,10 +23,48 @@ public void setup() {
...
-
         response = exampleApiService.getAllPosts();
 
         assertThat(response.statusCode())
+                .as("status code")
                 .isEqualTo(HttpStatus.SC_OK);
     }
+
+    @Test
+    public void WhenCallPostsEndpointThenJsonBodyIsCorrect() {
```
Comment (oliverCausley) [inline src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java]:
```suggestion
    public void whenCallPostsEndpointThenJsonBodyIsCorrect() {
```

File: src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java [hunk 3]
```diff
@@ -20,10 +23,48 @@ public void setup() {
...
+
+        softly.assertThat((int) jsonPath.get("id.size"))
+                .as("id size")
+                .isEqualTo(100);
+
+        softly.assertAll();
+    }
+
+    @Test
+    public void WhenCreateANewPostThenSuccessStatusReturned() {
```
Comment (oliverCausley) [inline src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java]:
```suggestion
    public void whenCreateANewPostThenSuccessStatusReturned() {
```

**Review by oliverCausley — CHANGES_REQUESTED** [review]
Your using a C# naming convention for test names is that intentional?

File: src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java [hunk 4]
```diff
@@ -20,10 +23,48 @@ public void setup() {
...
+
+        softly.assertThat(jsonPath.getString("id"))
+                .as("id content")
+                .contains("1", "2");
+
+        softly.assertThat((int) jsonPath.get("id.size"))
+                .as("id size")
+                .isEqualTo(100);
+
+        softly.assertAll();
```
Comment (oliverCausley) [inline src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java:48]: You could add a JUnit rule then you wouldn't have to call softly.assertAll()

Comment (ciaransweet) [reply src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java:48]: Not that it's a useful comment, but I think we had something similar to Softly inside the Tides tests somewhere, worth looking how we did those?

Comment (gregzealley) [reply src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java:48]: Couldn't see it.
On reflection, I think the current approach is the most easily understood and as this is a project to be used by those new to Java, that's a prime consideration.

File: src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java (hunk 3 above)
Comment (gregzealley) [reply src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java]: Doh, fixed.

File: src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java (hunk 2 above)
Comment (gregzealley) [reply src/test/java/uk/gov/ukho/apitest/tests/ExampleApiTest.java]: Doh, fixed.

**Review by oliverCausley — APPROVED** [review]
Looks good to me

**Pull request merged**